src/main/java/com/watch/shop/app/model/Watch.java

```diff
@@ -0,0 +1,20 @@
+package com.watch.shop.app.model;
+
+import lombok.*;
+
+import java.time.LocalDate;
+import java.util.UUID;
+
+@Data
+@AllArgsConstructor
+@Builder
+@EqualsAndHashCode(of = "id")
+public class Watch {
```

Comment:
In OOP related task or when you see about entity hierarchy often Abstract class is used , in such way in Interview tasks you could show that you are know and use OOP, in such case
we indeed have WatchType and should have this field but you can also introduce hierarchy that differs by few fields

SmartWatches have operationSystem enum, MechanicalWatch hasTurbillon boolean field etc, just introduce 1-2 additional fields.

Lombok Builder could be used as well while using inheritance